File: 3656-minimum-number-of-operations-to-make-elements-in-array-distinct/3656-minimum-number-of-operations-to-make-elements-in-array-distinct.java
class Solution {
    public int minimumOperations(int[] nums) {
        Set<Integer> seen = new HashSet<>();

        for (int i = nums.length - 1; i >= 0; i--) {
            int num = nums[i];
            if (seen.contains(num)) {
                return (i + 3) / 3;
            }
            seen.add(num);
        }

        return 0;
    }
}

File: 3656-minimum-number-of-operations-to-make-elements-in-array-distinct/3656-minimum-number-of-operations-to-make-elements-in-array-distinct.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var minimumOperations = function (nums) {
    const mp = new Map();
    let ans = 0;

    for (let num of nums) {
        mp.set(num, (mp.get(num) || 0) + 1);
    }

    if ([...mp.values()].every((count) => count === 1)) return 0;
    if (nums.length < 3) return 1;

    let i = 0;

    while (i < nums.length) {
        for (let j = i; j < i + 3 && j < nums.length; j++) {
            const val = nums[j];
            mp.set(val, mp.get(val) - 1);
            if (mp.get(val) === 0) {
                mp.delete(val);
            }
        }

        ans++;
        i += 3;

        if ([...mp.values()].every((count) => count === 1)) {
            return ans;
        }
    }

    return ans;
};
